.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,38 @@
+name: Build and test rmlint (master)
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      RM_TS_DIR: "/tmp/rmlint-unit-testdir"
+    steps:
+      - uses: actions/checkout@v4
+      - name: "Prepare build environment"
+        run: |
+          sudo apt update
+          sudo apt install -y --no-install-recommends \
+            scons python3-sphinx gettext python3-setuptools \
+            libblkid-dev libelf-dev libglib2.0-dev libjson-glib-dev \
+            clang python3-pip python3-cffi python3-dev libffi-dev
+          pip3 install -r tests/requirements.txt
+      - name: "Build"
+        # Todo: eventually run tests with valgrind (RM_TS_USE_VALGRIND)
+        # Todo enable slow tests in pytest
+        run: |
+          scons config
+          scons VERBOSE=1 DEBUG=1 O=release
+      - name: "Prepare test environment"
+        # Based off https://github.com/LeastAuthority/mount-tmpfs-action
+        run: |
+          sudo mkdir "${RM_TS_DIR}"
+          sudo mount -o size=14G,nr_inodes=0 -t tmpfs tmpfs "${RM_TS_DIR}"
```

Comment:
why custom temporary partition needed?  default one is too small? might then be worth adding a comment describing the rationale for this.

Reply:
I'm not sure about what the default tmpfs partition is on github runners, although usually it is half the ram (in this case, 8G). When running the tests locally I did not only ran out of space but also ran out of inodes on some of the heavier tests, so that's why I added `nr_inodes=0`. 

I also ran the whole suite locally with a Gen 4 NVMe drive and even then things were noticeably slower. I saw usage of around 8-9G. For some reason when mounting it with unlimited inodes it does not report the total so I don't have that number right now. 

I added a comment in the job file with some info. As a side note, I decreased the amount to 12G so that there's some leeway if someone goes nuts with the tests in the future :)